index.ts

```diff
@@ -0,0 +1,572 @@
+  }
+}> {}
+
+interface JSONMove extends Nullable<Data & CommonMove<ID, ID, ID>> {}
+interface Move extends DeepReadonly<Data & CommonMove<Item, Move, Type> & {
+  kind: 'Move';
+
+  noMetronome: Move[];
+  secondaries: SecondaryEffect[];
+
+  toJSON(): JSONMove;
+}> {}
+
+interface Moves {
+  get(id: ID): Move | undefined;
+  [Symbol.iterator](): IterableIterator<Move>;
+}
+
+// ---------- species.ts ----------
+
+type GenderName = 'M' | 'F' | 'N';
+type UsageTierName = 'OU' | 'UU' | 'RU' | 'NU' | 'PU';
+type TierName = UsageTierName | 'Uber' | 'UUBL' | 'RUBL' | 'NUBL' | 'PUBL';
+
+interface SpeciesAbilities<AbilityT> extends DeepReadonly<{
+  0: AbilityT;
+  1?: AbilityT;
+  H?: AbilityT;
+  S?: AbilityT;
+}> {};
+
+interface CommonSpecies<AbilityT, ItemT, MoveT, SpeciesT, TypeT> {
+  num: number;
+  abilities: SpeciesAbilities<AbilityT>;
+  baseForme?: SpeciesT;
+  baseSpecies?: SpeciesT;
+  baseStats: StatsTable;
+  battleOnly?: boolean;
+  cosmeticForms?: SpeciesT[];
+  doublesTier?: TierName;
+  eventOnly?: boolean;
+  evos?: SpeciesT[];
+  formeLetter?: string;
+  forme?: string;
+  isMega?: boolean;
+  isPrimal?: boolean;
+  isTotem?: boolean;
+  maleOnlyHidden?: boolean
+  maxHP?: number;
+  nfe: boolean;
+  otherFormes?: SpeciesT[];
+  prevo?: SpeciesT;
+  requiredAbility?: AbilityT;
+  requiredItems?: ItemT[];
+  requiredMove?: MoveT;
+  tier?: TierName;
+  types: TypeT[];
+  unreleasedHidden?: boolean
+  weightkg: number;
+}
+
+interface JSONSpecies extends Nullable<JSONData & CommonSpecies<ID, ID, ID, ID, ID>> {}
+interface Species extends DeepReadonly<Data & CommonSpecies<Ability, Item, Move, Species, Type> & {
+  type 'Species';
+
+  cosmeticForms: Species[];
+  otherFormes: Speces[];
+  requiredItems: Item[];
+
+  toJSON(): JSONSpecies;
+}> {}
+
+interface SpeciesSummaries {
+  getName(name: string): string;
+  get(id: ID): Species | undefined;
+  [Symbol.iterator](): IterableIterator<Species>;
+}
+
+// ---------- species-details.ts ----------
+
+// TODO: create EggGroup and Color enum types?
+
+type EvoType = 'Trade' | 'Stone' | 'LevelMove' | 'LevelExtra' | 'LevelFriendship' | 'LevelHold';
+
+interface CommonEventInfo<AbilityT, MoveT, NatureT> {
+  generation: Generation;
+  level?: number;
+  shiny?: 'Always' | 'Never' | 'Possible';
+  gender?: GenderName;
+  nature?: NatureName;
+  ivs?: Partial<StatsTable>;
+  perfectIVs?: number;
+  isHidden?: boolean;
+  abilities?: AbilityT[];
+  maxEggMoves?: number;
+  moves?: MoveT[];
+  pokeball?: ID;
+  from?: string;
+}
+
+interface JSONEventInfo extends Nullable<CommonEventInfo<ID, ID, NatureName>> {}
+interface EventInfo extends DeepReadonly<CommonEventInfo<Ability, Move, Nature>> {}
+
+type MoveSource = string;
+
+interface CommonSpeciesDetails<EventInfoT, ItemT, MoveT> {
+  canHatch?: boolean;
+  color: string; // Color
+  eggGroups?: string[]; // EggGroup
+  encounters?: EventInfoT[]
+  eventPokemon?: EventInfoT[];
+  evoDetails?: {
+    type: EvoType;
+    condition?: string;
```

Comment:
my first thought was "make an enum" but this appears to be a description, maybe use same `desc` naming convention?

Reply:
I'm pretty sure it's an enum...

Reply:
is stuff like `'near a special magnetic field'` something we can expect people to switch on?